Core/src/TupleObjects.cpp

```diff
@@ -81,12 +81,13 @@ TupleJet::TupleJet(const ntuple::Event& _event, size_t _jet_id)
 
 const LorentzVectorE& TupleJet::p4() const { return event->jets_p4.at(jet_id); }
 bool TupleJet::PassPuId(DiscriminatorWP wp) const {
+    analysis::DiscriminatorIdResults jet_pu_id(event->jets_pu_id.at(jet_id));
     if(wp == DiscriminatorWP::Loose)
```

Comment:
- simply `return jet_pu_id.Passed(wp);`
- also add method GetPuId that returns DiscriminatorIdResults and use it here and in EventInfoBase::SelectJets